lib/Raven/Client.php

```diff
@@ -112,13 +119,27 @@ class Client
     protected $config;
 
     /**
+     * @var FlexibleHttpClient The HTTP client
+     */
+    private $httpClient;
+
+    /**
+     * @var RequestFactory The PSR-7 request factory
+     */
+    private $requestFactory;
+
+    /**
      * Constructor.
      *
-     * @param Configuration $config The client configuration
+     * @param Configuration      $config         The client configuration
+     * @param FlexibleHttpClient $httpClient     The HTTP client
+     * @param RequestFactory     $requestFactory The PSR-7 request factory
      */
-    public function __construct(Configuration $config)
+    public function __construct(Configuration $config, FlexibleHttpClient $httpClient, RequestFactory $requestFactory)
```

Comment:
You should use only HttpAsyncClient if you are required to support it. You leave it to the user to wrap their non async client into a flex client.